AI/Nullkiller/AIGateway.cpp

```diff
@@ -790,14 +790,29 @@ bool AIGateway::makePossibleUpgrades(const CArmedInstance * obj)
 	{
 		if(const CStackInstance * s = obj->getStackPtr(SlotID(i)))
 		{
-			UpgradeInfo ui;
-			myCb->fillUpgradeInfo(obj, SlotID(i), ui);
-			if(ui.oldID != CreatureID::NONE && nullkiller->getFreeResources().canAfford(ui.cost[0] * s->count))
+			UpgradeInfo ui(s->getId());
```

Comment:
No need to save on variable name length. UI is also very unclear, since it often stands for "user interface"
```suggestion
			UpgradeInfo upgradeInfo(s->getId());
```